src/DIRAC/WorkloadManagementSystem/Utilities/JobParameters.py

```diff
@@ -185,3 +185,50 @@ def getNumberOfJobProcessors(jobID):
     return numberOfProcessors
 
   return getNumberOfProcessors()
+
+
+def getGPUs(siteName=None, gridCE=None, queue=None):
+  """ Gets GPUs on a certain CE/queue/node (what the pilot administers)
+
+      The siteName/gridCE/queue parameters are normally not necessary.
+
+      Tries to find it in this order:
+      1) from the /Resources/Computing/CEDefaults/GPUs (which is what the pilot might fill up)
+      2) if not present looks in CS for "%dGPUs" Queue or CE Tag
+      3) return 0
+  """
+
+  # 1) from /Resources/Computing/CEDefaults/GPUs
+  gLogger.info("Getting GPUs from /Resources/Computing/CEDefaults/GPUs")
+  gpus = gConfig.getValue('/Resources/Computing/CEDefaults/GPUs', 0)
+  if gpus:
+    return gpus
+
+  if not siteName:
+    siteName = gConfig.getValue('/LocalSite/Site', '')
+  if not gridCE:
+    gridCE = gConfig.getValue('/LocalSite/GridCE', '')
+  if not queue:
+    queue = gConfig.getValue('/LocalSite/CEQueue', '')
+  if not (siteName and gridCE and queue):
+    gLogger.error("Could not find NumberOfProcessors: missing siteName or gridCE or queue. Returning '0'")
+    return 0
+
+  # 2) looks in CS for tags
+  gLogger.info("Getting number of GPUs" "from tags for %s: %s: %s" % (siteName, gridCE, queue))
+  # Tags of the CE
+  tags = fromChar(gConfig.getValue('/Resources/Sites/%s/%s/CEs/%s/Tag' % (siteName.split('.')[0], siteName, gridCE),
+                                   ''))
+  # Tags of the Queue
+  tags += fromChar(gConfig.getValue('/Resources/Sites/%s/%s/CEs/%s/Queues/%s/Tag' % (siteName.split('.')[0],
+                                                                                     siteName,
+                                                                                     gridCE, queue),
+                                    ''))
+  for tag in tags:
```

Comment:
I think we should avoid this kind of tags in the CS and translate to tags only in the WMS internal machinery